v-next/hardhat-viem/src/internal/chains.ts

```diff
@@ -0,0 +1,127 @@
+import type { TestClientMode } from "../types.js";
+import type { EthereumProvider } from "@ignored/hardhat-vnext/types/providers";
+import type { Chain as ViemChain } from "viem";
+
+import { HardhatError } from "@ignored/hardhat-vnext-errors";
+import { extractChain } from "viem";
+import * as chainsModule from "viem/chains";
+import { hardhat, anvil } from "viem/chains";
+
+/* eslint-disable-next-line @typescript-eslint/consistent-type-assertions
+-- TODO: this assertion should not be necessary */
+const chains = Object.values(chainsModule) as ViemChain[];
+
+const chainIdCache = new WeakMap<EthereumProvider, number>();
+const isHardhatNetworkCache = new WeakMap<EthereumProvider, boolean>();
+const isAnvilNetworkCache = new WeakMap<EthereumProvider, boolean>();
+
+const HARDHAT_METADATA_METHOD = "hardhat_metadata";
+const ANVIL_NODE_INFO_METHOD = "anvil_nodeInfo";
+
+export async function getChain(provider: EthereumProvider): Promise<ViemChain> {
+  const chainId = await getChainId(provider);
+
+  const chain = extractChain({
+    chains,
+    id: chainId,
+  });
+
+  if (isDevelopmentNetwork(chainId) || chain === undefined) {
```

Comment:
I think this is probably too restrictive. I think it's better to do a check like:

```
if (hardhat network) {


}

if (anvil) {

}

// maybe here check if chain id is 1337 or 31337

```

The reason is that when forking people tend to change the chainid